7_react/react1/week2/react-app/src/components/List.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState } from "react";
+import { ListItem } from "./ListItem";
+
+export function List() {
+  const [list, setList] = useState(initialList);
+
+  useEffect(() => {
+    console.log("something");
+  }, [list]); // dependency on list changes
+
+  useEffect(() => {
+    console.log("hello");
+  }, []); // empty dependency array = only when the component mounts;
+
+  const addRandomTask = () => {
+    const newTask = {
+      id: list.length + 1,
+      description: `Random task ${list.length + 1}`,
+      done: false,
+    };
+    setList((previousState) => [...previousState, newTask]);
+  };
+
+  const markAsDone = (id) => {
+    setList((previousState) =>
+      previousState.map((task) =>
+        task.id === id ? { ...task, done: !task.done } : task
+      )
+    );
+  };
```

Comment:
Your app looks great, and you can both add and remove tasks just as expected. The user can't wait to use your app. However, after some time one user reports that the app sometimes behaves strange after she added and removed several items. It even shows duplicated items that she cannot individually mark as completed. 
![image](https://github.com/HackYourFuture-CPH/class26-homework/assets/97097933/776a942f-a559-429f-8927-17b921ab2803)

What could be the issue?

Hint, check the console for errors